GPU/Vulkan/FramebufferVulkan.cpp

```diff
@@ -937,112 +942,10 @@ void FramebufferManagerVulkan::DestroyAllFBOs() {
 	bvfbs_.clear();
 }
 
-void FramebufferManagerVulkan::FlushBeforeCopy() {
-	// Flush anything not yet drawn before blitting, downloading, or uploading.
-	// This might be a stalled list, or unflushed before a block transfer, etc.
-
-	// TODO: It's really bad that we are calling SetRenderFramebuffer here with
-	// all the irrelevant state checking it'll use to decide what to do. Should
-	// do something more focused here.
-	SetRenderFrameBuffer(gstate_c.IsDirty(DIRTY_FRAMEBUF), gstate_c.skipDrawReason);
-	if (!draw_->GetNativeObject(Draw::NativeObject::CURRENT_RENDERPASS))
-		Crash();
-	drawEngine_->Flush();
-}
-
 void FramebufferManagerVulkan::Resized() {
 	FramebufferManagerCommon::Resized();
 
 	if (UpdateSize()) {
 		DestroyAllFBOs();
 	}
 }
-
-bool FramebufferManagerVulkan::GetFramebuffer(u32 fb_address, int fb_stride, GEBufferFormat format, GPUDebugBuffer &buffer, int maxStride) {
-	// TODO: Doing this synchronously will require stalling the pipeline. Maybe better
-	// to do it callback-style?
```

Comment:
This is probably still a good idea for the "new world", as long as we're careful to ensure it doesn't hang when GE execution is stalled because the debugger has stalled it.

In practice, there are a few main use cases:
 * GE debugging (in practice async will not matter because it will stall anyway.)
 * Video file recording (would probably be great if it was async.)
 * Screenshots (benefit slightly from async.)
 * Save state screenshots (could probably be async but need to manage the stall.)

-[Unknown]

Reply:
Yup, it still would be nice. Adapted your comment here into a new comment on FramebufferCommon's generic implementation.